python/paddle/fluid/layers/nn.py

```diff
@@ -3250,3 +3251,25 @@ def one_hot(input, depth):
         attrs={'depth': depth},
         outputs={'Out': one_hot_out})
     return one_hot_out
+
+
+def global_step_counter():
+    """
+    Return the run counter of the main program, which is started with 1.
+    Returns(Variable): The global run counter.
+    """
+    helper = LayerHelper('global_step_counter')
+    counter_name = '@STEP_COUNTER@'
+    counter, is_new_var = helper.create_or_get_global_variable(
+        name=counter_name, dtype='int64', shape=[1], persistable=True)
+    if is_new_var:
+        helper.set_variable_initializer(
+            counter, initializer=Constant(
+                value=0, force_cpu=True))
+        helper.main_program.global_block().prepend_op(
+            type='increment',
```

Comment:
Maybe it's better to get a global counter and explicitly increase it?

global_step = get_or_create_global_step()
global_step += 1

global step needs to be a mutable thing, maybe a non-trainable parameter.

Reply:
1. the global step should not be a parameter. A parameter of a model should affect the output of the model. For example, `y=f(W; x)`, W is a parameter iif it will change the model behaviour.

2. The global step is a variable will be shared by `log_every_n`, `learning_rate_decays`, etc. Instead of creating several variables, here we just create a special global variable to record how many times the program has run.